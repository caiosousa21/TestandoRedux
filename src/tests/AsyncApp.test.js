import React from 'react';
import {AsyncApp, mapStateToProps} from '../containers/AsyncApp';
import Enzyme, { shallow, mount } from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() })

describe('index', () => {
    it('Renderizando AsyncApp', () => {
        const mockFn = jest.fn();
        const wrapper = shallow(<AsyncApp selectedSubreddit='' posts={[]} isFetching='' dispatch={mockFn} value=''/>);
        expect(wrapper.exists()).toBe(true);
    })
    it('MapStateToProps()', () => {
        const initialState={
            selectedSubreddit:'caio',
            postsBySubReddit:{
                selectedSubreddit:'caio',
                posts:[],
                isFetching:true
            }
        }
        expect(mapStateToProps(initialState)).toBe();
    })
    it('Testando interface com snapshots', () => {
        const mockFn = jest.fn();
        const tree = renderer.create(<AsyncApp selectedSubreddit='' posts={[]} isFetching='' dispatch={mockFn} value=''/>).toJSON();
        expect(tree).toMatchSnapshot()
    })
})