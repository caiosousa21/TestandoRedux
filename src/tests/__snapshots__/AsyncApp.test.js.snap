// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`index Testando interface com snapshots 1`] = `
<div>
  <span>
    <h1>
      
    </h1>
    <select
      onChange={[Function]}
      value=""
    >
      <option
        value="reactjs"
      >
        reactjs
      </option>
      <option
        value="frontend"
      >
        frontend
      </option>
    </select>
  </span>
  <p>
    <button
      onClick={[Function]}
    >
      Refresh
    </button>
  </p>
  
  <h2>
    Empty
  </h2>
</div>
`;
